[30기 한영현] MISSION4까지 완료

younghyunHan opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
1. 로그인 페이지 레이아웃 구현
2. ID, PW입력 시 로그인 버튼 활성화 기능
3. 메인 페이지 레이아웃 구현
4. 댓글 내용 입력 후 Enter press, 혹은 게시 버튼 클릭 시 댓글 추가 기능 
5. 리팩토링

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 로그인 페이지
- keyup 이벤트, 삼항연산자를 활용해 로그인 버튼 활성화 ( 연한 파란색 -> 파란색 )

2. 메인 페이지
- keyup 이벤트, createElement를 활용해 댓글 기능 구현
- overflow:scroll를 통해 스크롤바 구현 ( ::-webkit-scrollbar, display :none과 함께 사용 )
- innerText은 문자열 입력, innerHTML은 HTML문서 서식 활용 가능

3. 리팩토링
- div태그 중 불필요한 태그 제거 및 시맨틱태그로 수정
- css속성 우선순위에 따른 css파일 정렬


<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
1. 로그인 페이지
- window.addEventListener("keyup", function () : 윈도우 창에서 키 값이 입력되면 함수 실행
- doucument.body 쓰면 안됨. getElementsByClassName, getElementsById가 이미 body를 포함하고 있음.
- 함수 밖에서 변수를 선언할 때 value를 붙이면 로딩했을 떄 값이 담김.

2. 메인 페이지
- img태그 속성 순서 alt > class > src


<br />

## :: 기타 질문 및 특이 사항

yojuyoon commented on students/16th/moonkyuchan/js/login.js:
```diff
@@ -0,0 +1,19 @@
+  let idValue = inputId.value;
+  let passwordValue = inputPassword.value;
+
+  if(idValue.length>0 && passwordValue.length>0){
```
```suggestion
  if(idValue.length && passwordValue.length){
```
`>0`은 생략 가능합니다!

yojuyoon commented on students/16th/moonkyuchan/js/login.js:
```diff
@@ -0,0 +1,19 @@
+
+  if(idValue.length>0 && passwordValue.length>0){
+    inputBtn.style.cursor = "pointer";
+    inputBtn.style.backgroundColor = "#0069c0";
```
스타일링은 직접적으로 js에서 해주는 것 보다 기능 분리를 위해 classList를 활용해서 변경해보세요! 
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

yojuyoon commented on students/16th/moonkyuchan/styles/login.css:
```diff
@@ -0,0 +1,70 @@
+
+*{
+    box-sizing: border-box;
+}
```
해당 속성은 초기화 속성이므로 추후 reset.css에서 적용해주시면 됩니다!

yojuyoon commented on students/16th/moonkyuchan/styles/login.css:
```diff
@@ -0,0 +1,70 @@
+    justify-content: center;
+    align-items: center;
+    background-color: #e0e0e0;
+    width : 100vw;
```
body자체에 width와 height값을 주신 이유가 있을까요?

yojuyoon commented on students/16th/moonkyuchan/styles/login.css:
```diff
@@ -0,0 +1,70 @@
+    padding: 80px 0px 50px 0px;
+    
+}
+#firstInput{
+    height: 40px;
+    margin : 0px 0px 10px 0px;
+    border-radius: 3px;
+    border : 1px solid #c1c1c1;
+    background-color: #f5f5f5;
+    padding-left: 10px;
+    
+}
+#secondInput{
+    height: 40px;
+    margin : 0px 0px 20px 0px;
+    border-radius: 3px;
+    border : 1px solid #c1c1c1;
+    background-color: #f5f5f5;
+    padding-left: 10px;
+}
```
동일한 속성을 같은 태그에 적용중이네요! 중복 속성은 합쳐서 할당 가능합니다!

yojuyoon reviewed (commented):
규찬님! 리뷰 내용 확인 부탁드립니다! 🤗

younghyunHan committed:
Add: login, main 파일 추가
- login id, pw 입력받기
- 입력받은 값에 따라 button 색상 병경
- 버튼 클릭 시 메인 페이지로 이동

younghyunHan committed:
- [Add] main.html  레이아웃 & login.html 레이아웃
 - login.js에 삼항 연산자로 login페이지  로그인 버튼 활성화
 - main.js에 addEventListener를 활용하여 댓글 기능 추가

younghyunHan committed:
Add main페이지, main.css 리팩토링
- 시맨틱 태그 및 불필요한 태그 제거
- css 속성 우선순위에 따른 정렬 수정 중

younghyunHan committed:
[Modify] main.css 리팩토링
- CSS 속성 순서 우선순위대로 수정

younghyunHan committed:
- [Modify] common.css  body태그 height 수정
 - common.css에 height를 불필요하게  100vh로 지정한 부분 수정함.

younghyunHan committed:
- [Modify]  main.css body태그 height 삭제
